### PR TITLE
raftstore: force compact with gentleness

### LINKED ISSUE #11989: After a tikv fault (more than 30 minutes) is restored, the raft log lag has been chasing
### What version of TiKV are you using?
[2022/02/16 19:56:48.223 +08:00] [INFO] [client.go:338] ["Cluster version information"] [type=pd] [version=5.5.0-nightly] [git_hash=f4b0b3187aa1e044f72b3cbe9048f29d1878c584]
[2022/02/16 19:56:48.223 +08:00] [INFO] [client.go:338] ["Cluster version information"] [type=tikv] [version=5.4.0-alpha] [git_hash=8f4f71ea655c5ba794fd2f1780e98e6c3831bc16]
[2022/02/16 19:56:48.223 +08:00] [INFO] [client.go:338] ["Cluster version information"] [type=tidb] [version=5.5.0-nightly] [git_hash=9659b93250e7f08e33f76e59844835927e3d39d8]

### What operating system and CPU are you using?
8 core 16G

### Steps to reproduce
run tpcc or sysbenc insert 
inject one tikv fault and recover （more than 30min）

### What did you expect?
raft log lag can becomes smaller

### What did happened?
raft log lag becomes more and more
middle_img_v2_bc59e1bf-5726-4bc5-ba67-0f5f8da3597g
middle_img_v2_0a248809-5e85-4b36-b7c7-55ed955f4eeg
middle_img_v2_b73c4645-e7a1-4050-9139-bd4546d04f0g

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

### What is changed and how it works?

Issue Number: Close #11989

What's Changed:

Only delete the oldest 25% raft logs during force compact.

### Check List 

Tests

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore/src/store/fsm/peer.rs [hunk 1]
```diff
@@ -4264,14 +4264,15 @@ where
         let mut total_gc_logs = 0;
 
         let first_idx = self.fsm.peer.get_store().first_index();
+        let last_idx = self.fsm.peer.get_store().last_index();
```
Comment (Connor1996) [inline components/raftstore/src/store/fsm/peer.rs:4267]: there is a `last_idx` on L4228

**Review by Connor1996 — COMMENTED** [review]
rest LGTM

**Review by Connor1996 — APPROVED** [review]
Please paste the test result in PR description

**Comment by tabokie** [comment]
Honestly I'm not sure why the lag started shrinking even before tikv-2 went back online.

image
image

**Comment by Connor1996** [comment]
> 

After 30min, PD begins to remove the down peers and add to other stores, so the log lag is reduced.

**Review by 5kbpers — APPROVED** [review]

**Pull request merged**